### PR TITLE
refactor: simplify `rebuildShares`

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/rsmt2d/issues/196

Since [`Decode`](https://github.com/rootulp/rsmt2d/blob/1f1904acc114b41dff838ad1d530feacc1d9f199/leopard.go#L46-L54) returns original + parity shares, there is no need to separately rebuild the parity shares. This PR removes an unnecessary conditional inside `rebuildShares` which let us also remove a param to that function. After the refactor, a few helper methods were no longer used so they were also removed.

### CODE REVIEW

File: extendeddatacrossword.go
Comment (evan-forbes) [inline extendeddatacrossword.go:249]: do we need this function at all if its just decoding?

**Review by evan-forbes — APPROVED** [review]
ahh okay so the encode and decode abstractions were (are?) leaky

File: extendeddatacrossword.go
Comment (staheri14) [inline extendeddatacrossword.go:262]: [Question] I suppose there have not been any tests associated with this part of the code, as no test file has been updated in this PR, right?

File: extendeddatacrossword.go
Comment (staheri14) [inline extendeddatacrossword.go:264]: [Question] The only potential issue could arise if the parity portion is incomplete and the rebuilt extended shares obtained from `rebuiltShares, err := eds.codec.Decode(shares)` (the second half of  `rebuiltShares`) and `rebuiltExtendedShares, err := eds.codec.Encode(rebuiltShares[0:eds.originalDataWidth])` turn out to be different. Have we conducted any tests to verify this?

In the original code, we initially decode the entire row, and if the extended part (before decoding) is incomplete, we discard that part from the decoded version and replace it with the encoding result. However, in the new version, we retain whatever the decode function provides us with. If we are confident that these two have the same result, then everything looks good to me! (I think they should be the same)

**Review by staheri14 — APPROVED** [review]
LGTM! left two questions, but they are non-blocking as long as we are certain about them.

File: extendeddatacrossword.go
Comment (rootulp) [reply extendeddatacrossword.go:249]: good question, I can explore deleting it entirely.

File: extendeddatacrossword.go
Comment (rootulp) [reply extendeddatacrossword.go:262]: codecov claims these lines were covered: https://app.codecov.io/gh/celestiaorg/rsmt2d/blob/master/extendeddatacrossword.go

I assume via https://github.com/rootulp/rsmt2d/blob/1f1904acc114b41dff838ad1d530feacc1d9f199/extendeddatacrossword_test.go#L21

File: extendeddatacrossword.go
Comment (rootulp) [reply extendeddatacrossword.go:264]: > The only potential issue could arise if the parity portion is incomplete

I don't expect this to happen because Decode calls Reconstruct [here](https://github.com/rootulp/rsmt2d/blob/1f1904acc114b41dff838ad1d530feacc1d9f199/leopard.go#L52) and the godoc for Reconstruct claims:

> 	// If there are too few shards to reconstruct the missing
	// ones, ErrTooFewShards will be returned.

Ref: https://github.com/klauspost/reedsolomon/blob/523164698be98f1603cf1235f5a1de17728b2091/reedsolomon.go#L56-L57

File: extendeddatacrossword.go
Comment (rootulp) [reply extendeddatacrossword.go:249]: TBH I'm only creating a new issue rather than tackling in this PR b/c I don't want to dismiss existing approvals 😞 https://github.com/celestiaorg/rsmt2d/issues/199

**Pull request merged**
